File: mediapicker/src/main/java/vn/tungdx/mediapicker/activities/BaseFragment.java
package vn.tungdx.mediapicker.activities;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;

import vn.tungdx.mediapicker.imageloader.MediaImageLoader;

/**
 * Created by TungDX
 */
public class BaseFragment extends Fragment {
    protected Context mContext;
    protected MediaImageLoader mMediaImageLoader;

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        FragmentHost host = (FragmentHost) activity;
        mMediaImageLoader = host.getImageLoader();
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mContext = getActivity();
    }
}

File: mediapicker/src/main/java/vn/tungdx/mediapicker/activities/BaseFragment.kt
package vn.tungdx.mediapicker.activities

import android.app.Activity
import android.content.Context
import android.os.Bundle

import androidx.fragment.app.Fragment
import vn.tungdx.mediapicker.imageloader.MediaImageLoader

/**
 * Created by TungDX
 */
open class BaseFragment : Fragment() {

    protected var mContext: Context? = null

    protected lateinit var mMediaImageLoader: MediaImageLoader

    override fun onAttach(activity: Activity?) {
        super.onAttach(activity)
        val host = activity as FragmentHost?
        mMediaImageLoader = host!!.imageLoader
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        mContext = activity
    }
}
